File: src/main/java/no/novari/flyt/egrunnerverv/gateway/exception/ArchiveResourceNotFoundException.java
package no.novari.flyt.egrunnerverv.gateway.exception;

import no.novari.flyt.egrunnerverv.gateway.slack.SlackAlertService;
import no.novari.flyt.instance.gateway.exception.AbstractInstanceRejectedException;

public class ArchiveResourceNotFoundException extends AbstractInstanceRejectedException {

    private static final String MESSAGE_TEMPLATE = "No archive resource found for saksansvarligEpost='%s'";

    public ArchiveResourceNotFoundException(String email, SlackAlertService slackAlertService) {
        super(formatMessage(email));
        slackAlertService.sendMessage(formatMessage(email));
    }

    private static String formatMessage(String email) {
        return String.format(MESSAGE_TEMPLATE, email);
    }
}

File: src/main/kotlin/no/novari/flyt/egrunnerverv/gateway/exception/ArchiveResourceNotFoundException.kt
package no.novari.flyt.egrunnerverv.gateway.exception

import no.novari.flyt.egrunnerverv.gateway.slack.SlackAlertService
import no.novari.flyt.gateway.webinstance.exception.AbstractInstanceRejectedException

class ArchiveResourceNotFoundException(
    email: String,
    slackAlertService: SlackAlertService,
) : AbstractInstanceRejectedException(formatMessage(email)) {
    init {
        slackAlertService.sendMessage(formatMessage(email))
    }

    companion object {
        private const val MESSAGE_TEMPLATE = "No archive resource found for saksansvarligEpost='%s'"

        private fun formatMessage(email: String): String = MESSAGE_TEMPLATE.format(email)
    }
}
